src/datasources/cache/redis.cache.service.ts

```diff
@@ -38,32 +40,39 @@ export class RedisCacheService
       return;
     }
 
+    const key = `${this.keyPrefix}${cacheDir.key}`;
+
     try {
-      await this.client.hSet(cacheDir.key, cacheDir.field, value);
-      await this.client.expire(cacheDir.key, expireTimeSeconds);
+      await this.client.hSet(key, cacheDir.field, value);
+      await this.client.expire(key, expireTimeSeconds);
     } catch (error) {
-      await this.client.hDel(cacheDir.key, cacheDir.field);
+      await this.client.hDel(key, cacheDir.field);
       throw error;
     }
   }
 
   async get(cacheDir: CacheDir): Promise<string | undefined> {
-    return await this.client.hGet(cacheDir.key, cacheDir.field);
+    const key = `${this.keyPrefix}${cacheDir.key}`;
+    return await this.client.hGet(key, cacheDir.field);
   }
 
   async deleteByKey(key: string): Promise<number> {
+    const keyWithPrefix = `${this.keyPrefix}${key}`;
     // see https://redis.io/commands/unlink/
-    const result = await this.client.unlink(key);
+    const result = await this.client.unlink(keyWithPrefix);
     await this.set(
-      new CacheDir(`invalidationTimeMs:${key}`, ''),
+      new CacheDir(`invalidationTimeMs:${keyWithPrefix}`, ''),
       Date.now().toString(),
       this.defaultExpirationTimeInSeconds,
     );
     return result;
   }
 
   async deleteByKeyPattern(pattern: string): Promise<void> {
-    for await (const key of this.client.scanIterator({ MATCH: pattern })) {
+    const patternWithPrefix = `${this.keyPrefix}${pattern}`;
```

Comment:
Nit: we could create a helper to prefix keys. It should work in all of these methods.

Reply:
I think it would also be beneficial to have a separator between the prefix and key for clarity. At the moment it's not clear where the prefix ends and the chain ID starts.